### PR TITLE
Issue #12 Patch: regex.inc

### PR DESCRIPTION
**Pull request by VasasA**
regex.inc can replace more functions: https://github.com/backdrop-contrib/coder_upgrade/issues/12

### CODE REVIEW

File: conversions/regex.inc [hunk 1]
```diff
@@ -13,6 +13,8 @@
  * Upgrades text using simple regular expressions.
  */
 function coder_upgrade_upgrade_regex_alter(&$file) {
+  global $_coder_upgrade_module_name;
+  $name=$_coder_upgrade_module_name;
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $name = $_coder_upgrade_module_name;
```

File: conversions/regex.inc [hunk 2]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
   $from[] = "/format_username/";
   $to[] = "user_format_name";
 
+  // theme_html_tag() has been renamed.
+  $from[] = "/theme_html_tag/";
+  $to[] = "theme_head_tag";
+
+  // hook_page_build() has been replaced: https://api.backdropcms.org/change-records/new-layout-module-implements-drag-drop-model-building-layouts
+  $from[] = "/".$name."_page_build/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = "/" . $name . "_page_build/";
```

File: conversions/regex.inc [hunk 3]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = $name."_preprocess_page";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $to[] = $name . "_preprocess_page";
```

File: conversions/regex.inc [hunk 4]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+
+  // hook_library() has been renamed.
+  $from[] = "/".$name."_library/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = "/" . $name . "_library/";
```

File: conversions/regex.inc [hunk 5]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = $name."_library_info";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $to[] = $name . "_library_info";
```

File: conversions/regex.inc [hunk 6]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+
+  // hook_library_alter() has been renamed.
+  $from[] = "/".$name."_library_alter/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = "/" . $name . "_library_alter/";
```

File: conversions/regex.inc [hunk 7]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = $name."_library_info_alter";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $to[] = $name . "_library_info_alter";
```

File: conversions/regex.inc [hunk 8]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+
+  // replace special sorting function: https://api.backdropcms.org/change-records/generic-sorting-function-replace-implementations-uasort
+  $from[] = '/uasort\(\$items, *'."'element_sort'\)/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = '/uasort\(\$items, *' . "'element_sort'\)/";
```

File: conversions/regex.inc [hunk 9]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = 'backdrop_sort($items, array("#weight"))';
+
+  // replace special sorting function 2
+  $from[] = '/uasort\(\$items, *'."'element_sort_title'\)/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = '/uasort\(\$items, *' . "'element_sort_title'\)/";
```

File: conversions/regex.inc [hunk 10]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = 'backdrop_sort($items, array("#title" => SORT_STRING))';
+
+  // replace special sorting function 3
+  $from[] = '/uasort\(\$items, *'."'drupal_sort'\)/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = '/uasort\(\$items, *' . "'drupal_sort'\)/";
```

File: conversions/regex.inc [hunk 11]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = 'backdrop_sort($items)';
+
+  // replace special sorting function 4
+  $from[] = '/uasort\(\$items, *'."'drupal_sort_title'\)/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = '/uasort\(\$items, *' . "'drupal_sort_title'\)/";
```

File: conversions/regex.inc [hunk 12]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = 'backdrop_sort($items, array("title" => SORT_STRING))';
+
+  // replace special sorting function 5
+  $from[] = '/uasort\(\$items, *'."'admin_bar_element_sort'\)/";
```
Comment (klonos) [inline conversions/regex.inc]:
```suggestion
  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
```

File: conversions/regex.inc [hunk 13]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false 
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
+  // change 'Drupal 7' to 'Backdrop CMS' throughtout.
```
Comment (klonos) [inline conversions/regex.inc]: This change should not be automated. Consider the following sentence for example:

> Backdrop CMS does things differently than Drupal 7.

...would become:

> Backdrop CMS does things differently than Backdrop CMS.

**Comment by docwilmot** [comment]
Apart from the coding standards suggestions @klonos made, I don't think these conversions can be mapped exactly as suggested. For example a `hook_page_build()` implementation cant just be renamed to `hook_preprocess_page()` as the two function would have been doing different things, and it would be entirely possible the D7 site would have had *both* functions. I'm similarly not sure about the `backdrop_sort()` switches, but havent checked. In all this still needs some work.

**Comment by VasasA** [comment]
@docwilmot  Please ignore and close my PR, and use the replacements only what are sure to work. For example: hook_library() -> hook_library_info()
https://api.backdropcms.org/change-records/hooklibrary-and-hooklibraryalter-have-been-renamed-hooklibraryinfo-and

File: conversions/regex.inc [hunk 14]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $from[] = "/Drupal 7/";
+  $to[] = "Backdrop CMS";
+
   // change Drupal to Backdrop throughtout.
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
```

File: conversions/regex.inc [hunk 15]
```diff
@@ -41,11 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  // replace special sorting function 5
+  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false 
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
+  // change 'Drupal 7' to 'Backdrop CMS' throughtout.
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
```

File: conversions/regex.inc [hunk 16]
```diff
@@ -41,12 +43,51 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
-  // change Drupal to Backdrop throughtout.
+  // change 'Drupal 7' to 'Backdrop CMS' throughtout.
+  $from[] = "/Drupal 7/";
+  $to[] = "Backdrop CMS";
+
   $from[] = "/drupal/";
```
Comment (bugfolder) [inline conversions/regex.inc:90]:
```suggestion
```

File: conversions/regex.inc [hunk 17]
```diff
@@ -41,12 +43,50 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false 
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
-  // change Drupal to Backdrop throughtout.
+  $from[] = "/Drupal 7/";
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
```

File: conversions/regex.inc [hunk 18]
```diff
@@ -41,12 +43,49 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false 
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
-  // change Drupal to Backdrop throughtout.
+  $to[] = "Backdrop CMS";
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
```

File: conversions/regex.inc [hunk 19]
```diff
@@ -41,12 +43,48 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false 
   // links.
   $from[] = "/drupal\.org/";
   $to[] = "mothershiporg";
 
-  // change Drupal to Backdrop throughtout.
+
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
  // change Drupal to Backdrop throughout.
```

File: conversions/regex.inc [hunk 20]
```diff
@@ -41,12 +43,48 @@ function coder_upgrade_upgrade_regex_alter(&$file) {
...
+  // replace special sorting function 4
+  $from[] = '/uasort\(\$items, *' . "'drupal_sort_title'\)/";
+  $to[] = 'backdrop_sort($items, array("title" => SORT_STRING))';
+
+  // replace special sorting function 5
+  $from[] = '/uasort\(\$items, *' . "'admin_bar_element_sort'\)/";
+  $to[] = 'backdrop_sort($items, array("#weight" => SORT_NUMERIC, "#title" => SORT_STRING))';
+
   // temporarily rename Drupal links so the other regexes dont create false
```
Comment (bugfolder) [inline conversions/regex.inc]:
```suggestion
  // temporarily rename Drupal links so the other regexes don't create false 
```

**Comment by bugfolder** [comment]
Closing this PR and making selected changes from it in PR https://github.com/backdrop-contrib/coder_upgrade/pull/67.

**Pull request closed without merging**